action_server/frontend/src/components/CancelButton.tsx

```diff
@@ -0,0 +1,44 @@
+import { FC, useMemo } from 'react';
+import { Button, Tooltip } from '@sema4ai/components';
+import { IconStop } from '@sema4ai/icons';
+import { cancelRun } from '~/lib/requestData';
+import { RunStatus } from '~/lib/types';
+
+type Props = {
+  runId?: string;
+  disabled?: boolean;
+  status: RunStatus;
+  requestId?: string;
+};
+
+export const CancelButton: FC<Props> = ({ runId, disabled, status, requestId }) => {
+  // Either runId or requestId must be provided.
+  if (!runId && !requestId) {
+    throw new Error('Either runId or requestId must be provided.');
```

Comment:
showing some kind of error state instead of throwing an exception would be less disruptive UX - minor though, as I'm assuming this state should not be reached at all

Reply:
Actually, this is a programming error -- it's more an assertion error that an invariant in the code is broken than something expected (so, a better UX should really be required).